flask_tutorial-code_review/main.py

```diff
@@ -0,0 +1,71 @@
+from flask import Flask, render_template, request, redirect, url_for, session, flash
+from flask_pymongo import PyMongo
+from datetime import datetime
+from flask_migrate import Migrate
+from config import DevelopmentConfig
+
+app = Flask(__name__, static_url_path='/static')
+app.config['MONGO_URI'] = 'mongodb://localhost:27017/yourdatabase'
+mongo = PyMongo(app)
+
+class Entry:
+    def __init__(self, user_id):
+        self.timestamp = datetime.utcnow()
+        self.user_id = user_id
+
+class User:
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
+@app.route('/')
+def login():
+    return render_template('login.html')
+@app.route('/index/', methods=['POST'])
+def index():
+    if request.method == 'POST':
+        username = request.form['nm']
+        password = request.form['password']
+
+        # Check if the user already exists
+        user = mongo.db.users.find_one({'username': username})
+
+        if user:
+            # If the user exists, update the password
+            mongo.db.users.update_one({'username': username}, {'$set': {'password': password}})
+        else:
+            # If the user doesn't exist, create a new user
+            new_user = {'username': username, 'password': password}
+            mongo.db.users.insert_one(new_user)
+
+            # Create a new entry for the user
+            entry = {'user_id': new_user['_id'], 'timestamp': datetime.utcnow()}  # Assuming _id is automatically generated
+            mongo.db.entries.insert_one(entry)
+
+        flash('Logged in successfully!')
+        session['username'] = username
```

Comment:
try having uid in session and not username for security and flexability reasons

Reply:
done